doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,680 @@
+`"minimizeToTray": true` set in their settings. If the user has that set to
+false, but would still like the tray, they can specify `"alwaysShowTrayIcon":
+true`. That will cause the tray icon to always be added to the system tray.
+
+There's not a combination of settings where the Terminal is "minimized to the
+tray", and there's _no tray icon visible_. We don't want to let users get into a
+state where the Terminal is running, but is totally hidden from their control.
+
+From a technical standpoint, the tray icon is managed similar to the global
+hotkey. The Monarch process is responsible for setting it up, and processing the
+messages. When a Monarch dies and a new process becomes the Monarch, then it
+will re-create the tray icon.
+
+## UI/UX Design
+
+To summarize, we're proposing the following set of settings:
+
+```jsonc
+{
+    "minimizeToTray": bool,
+    "alwaysShowTrayIcon": bool,
+    "actions": [
+        {
+            "keys": KeyChord,
+            "command": {
+                "action": "globalSummon",
+                "dropdownDuration": float,
+                "toggleVisibility": bool,
+                "monitor": "any"|"toCurrent"|"onCurrent"|int,
+                "desktop": "any"|"toCurrent"|"onCurrent"
+            }
+        },
+        {
+            "keys": KeyChord,
+            "command": {
+                "action": "quakeMode"
+            }
+        }
+    ]
+}
+```
+
+## Potential Issues
+
+<table>
+
+<tr>
+<td><strong>Compatibility</strong></td>
+<td>
+
+As part of this set of changes, we'll also be allowing the <kbd>Win</kbd> key in
+keybindings. Generally, the OS reserves the Windows key for its own shortcuts.
+For example, <kbd>Win+R</kbd> for the run dialog, <kbd>Win+A</kbd> for the
+Action Center, <kbd>Win+V</kbd> for the cloud clipboard, etc. Users will now be
+able to use the win key themselves, but they should be aware that the OS has
+"first dibs" on any hotkeys involving the Windows key.
+
+</td>
+</tr>
+<tr>
+<td><strong>Mixed elevation</strong></td>
+<td>
+
+Only one app at a time gets to register for global hotkeys. However, from the
+Terminal's perspective, unelevated and elevated windows will act like different
+apps. Each privilege level has its own Monarch. The two are unable to
+communicate across the elevation boundary.
+
+This means that if the user often runs terminals in both contexts, then only one
+will have the global hotkeys bound. The naive implementation would have the
+first elevation level "win" the keybindings.
+
+A different option would be to have elevated windows not register global hotkeys
+_at all_. I don't believe that there's any sort of security implication for
+having a global hotkey for an elevated window.
+
+A third option would be to have some sort of `"whenElevated": bool?` property
+for global hotkeys. This would explicitly enable a given hotkey for unelevated
+vs elevated windows.
+* `"whenElevated": null`: behave as normal - the first context level to run wins

```

Comment:
Do you intend to warn in some way that the hotkey couldn't be registered?

Reply:
> * If there are any other applications running that have already registered
>   hotkeys with `RegisterHotKey`, then it's possible that the Terminal's attempt
>   to register that hotkey will fail. If that should happen, then we should
>   display a warning dialog to the user indicating which hotkey will not work
>   (because it's already used for something else).